src/Plugin/Block/NodeShareBlock.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace Drupal\y_lb\Plugin\Block;
+
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Routing\CurrentRouteMatch;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides Share block.
+ *
+ * @Block(
+ *   id = "lb_node_share",
+ *   admin_label = @Translation("Share Block"),
+ *   category = @Translation("Common blocks")
+ * )
+ */
+class NodeShareBlock extends BlockBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * The route provider.
+   *
+   * @var \Drupal\Core\Routing\CurrentRouteMatch
+   */
+  protected $currentRouteMatch;
+
+  /**
+   * Cuurent node.
+   *
+   * @var \Drupal\node\Entity\Node
+   */
+  protected $node;
+
+  /**
+   * Constructs a new NodeShareBlock instance.
+   *
+   * @param array $configuration
+   *   The plugin configuration, i.e. an array with configuration values keyed
+   *   by configuration option name. The special key 'context' may be used to
+   *   initialize the defined contexts by setting it to an array of context
+   *   values keyed by context names.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Routing\RouteProviderInterface $route_provider
+   *   The route provider.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, CurrentRouteMatch $currentRouteMatch) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->currentRouteMatch = $currentRouteMatch;
+    $this->node = $currentRouteMatch->getParameter('node');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('current_route_match')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build() {
+    if (!$this->node) {
+      return [];
+    }
+
+    $url = urlencode($this->node->toUrl('canonical', ['absolute' => TRUE])->toString());
+
+    $build = [
+      '#links' => [
```

Comment:
Let's have $links as an array and define hook_alter in order to be able to change it in contribs.
To add alter hook in $module.api docs, please

Reply:
Last discussion we decided to implement Block configuration form to allow editors to input own links, however requirements have not been fully defined. Can we have both changes in the follow-up tickets or do you insist on the changes? I'd put the TODO ref for the future development.

Reply:
Adding hook_alter is one line of code, and no matter if we would have user entry or no, - we need alter for these kind of global things @andreyzb